### PR TITLE
[Deferred] GBuffer Normals format check

### PR DESCRIPTION
**Pull request by jonuuukas**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Not all android devices support RGBA8_SNorm, so adding a check for it to fix case 1288042

---
### Testing status
Manually tested on repro project on Google Pixel (which doesn't support it)
---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by kaychang-unity** [comment]
Thanks. the problem is we do use the whole [-1; 1] range to store normal vectors, so switching to R8G8B8A8_UNorm will cause graphical artefacts if the shader code is not fixed. If you look at the PackNormal/PackSmoothness/UnpackNormal/UnpackSmoothness functions:
https://github.com/Unity-Technologies/Graphics/blob/d066ddc120aa1d6b9ed46c388623cc2b6b19eca8/com.unity.render-pipelines.universal/ShaderLibrary/UnityGBuffer.hlsl#L55
Since we only know at runtime if the given Android platform supports R8G8B8A8_SNorm, you would need to add a shader variant for it.

I did consider always using R8G8B8A8_UNorm in all cases to make things simpler, but R8G8B8A8_SNorm offers 2 advantages:
- reduce artefact with terrain layers: the 5th layer of the terrain will blend gbuffer parameters, and the normals look more correct if the normals are stored in R8G8B8A8_SNorm. You can check this in UniversalGraphicsTests scene 038_Shaders_TerrainBlend and look at how specular is reflected on the terrain in forward vs deferred.
- Slightly cheaper to load/store normals. It does save some time on mobile platforms and Switch.

**Comment by jonuuukas** [comment]
@kaychang-unity - updated the PR as we spoke - forcing the option now for those platforms instead of just switching the format

File: com.unity.render-pipelines.universal/Runtime/DeferredLights.cs [hunk 1]
```diff
@@ -267,6 +267,10 @@ internal enum GBufferHandles
         internal int GBufferShadowMask { get { return UseShadowMask ? GBufferLightingIndex + 1 : -1; } }
         internal int GBufferSliceCount { get { return 4 + (UseRenderPass ? 1 : 0) + (UseShadowMask ? 1 : 0); } }
 
+        internal GraphicsFormat GBufferDefaultNormalsFormat = RenderingUtils.SupportsGraphicsFormat(GraphicsFormat.R8G8B8A8_SNorm, FormatUsage.Render)
```
Comment (kaychang-unity) [inline com.unity.render-pipelines.universal/Runtime/DeferredLights.cs:270]: This line has become unnecessary.

**Review by kaychang-unity — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/DeferredLights.cs (hunk 1 above)
Comment (jonuuukas) [reply com.unity.render-pipelines.universal/Runtime/DeferredLights.cs]: thanks for noticing :) fixed in dffbcd2

**Review by ellioman — APPROVED** [review]

**Comment by ellioman** [comment]
The Yamato Failures were all known issues not related to this PR plus the property being changed here does not have any testing.

**Comment by kaychang-unity** [comment]
On a side note, I recently found the same issue with SSAO after it has been modified to support deferred renderer.. We are now using GraphicsFormat.R8G8B8A8_SNorm to store the normal buffer, but it is not supported on some Adreno GPUs. Because SSAO code can run on GLES2, I cannot use accurateGBufferNormal option as bit operatations are not available in the shaders.
Instead, I am using GraphicsFormat.R16G16B16A16_Float as a fallback in that case.

**Pull request merged**